guides/messaging-high-availability-openshift.adoc

```diff
@@ -0,0 +1,296 @@
+
+Next, create a file named `broker.yaml` with the following content to deploy the AMQ 7 cluster:
+[source, yaml, options="nowrap"]
+----
+apiVersion: broker.amq.io/v1beta1
+kind: ActiveMQArtemis
+metadata:
+  name: amq-broker
+  application: amq-broker-app
+spec:
+  acceptors:
+    - name: acceptor
+      protocols: core,amqp
+      port: 61616
+      sslEnabled: false
+      enabledProtocols: TLSv1,TLSv1.1,TLSv1.2
+      needClientAuth: true
+      wantClientAuth: true
+      verifyHost: true
+      sslProvider: JDK
+      sniHost: localhost
+      expose: true
+      anycastPrefix: jms.queue.
+      multicastPrefix: /topic/
+  console:
+    expose: true
+  deploymentPlan:
+    journalType: nio
+    messageMigration: true
+    persistenceEnabled: true
+    requireLogin: false
+    size: 2
+    storage:
+      size: "1Gi"
+  upgrades:
+    enabled: false
+    minor: false
+----
+
+This Custom Resource (CR) file configures an AMQ 7 cluster with two brokers deployed as a StatefulSet. This setup ensures
+high availability because, in the event of a pod failure, the StatefulSet will automatically restart the failed pod.
+However, to maintain data integrity, Persistent Volumes must be configured to store the messaging journal. Without them,
+messages will be lost during restarts.
+
+Additionally, the `messageMigration: true` setting enables the graceful scaling down of AMQ 7 pods. This ensures that messages
+from the scaled-down node are migrated to another node in the cluster, preventing data loss.
+
+Run following command to deploy AMQ 7 on OpenShift:
+[source, bash, options="nowrap"]
+----
+oc create -f broker.yaml
+----
+
+Check that AMQ 7 brokers are in `Running` state by checking running pods:
+
+[source, bash ,options="nowrap"]
+----
+$ oc get pods
+NAME                                            READY   STATUS      RESTARTS   AGE
+amq-broker-ss-0                                  1/1     Running   0          35m
+amq-broker-ss-1                                  1/1     Running   0          36m
+...
+----
+
+== Build WildFly with a messaging application image using the WildFly Maven Plugin
+
+We're going to use the https://github.com/wildfly/quickstart/tree/main/remote-helloworld-mdb[WildFly Quickstart: remote-helloworld-mdb, window=_blank]
+to demonstrate how to build a trimmed WildFly server with a deployed messaging application. This quickstart leverages the WildFly Maven plugin
+to create a trimmed WildFly server and deploy the `remote-helloworld-mdb.war` application. It then produces a new container image based on
+the quay.io/wildfly/wildfly-runtime:latest[WildFly Runtime Image, window=_blank], incorporating the application.
```

Comment:
the URL here is missing "https://" so it doesn't render as a link.

Reply:
Nice catch, fixed. Thanks!